doc/source/reference/c-api/array.rst

```diff
@@ -3521,6 +3551,40 @@ member of ``PyArrayDTypeMeta_Spec`` struct.
    force newly created arrays to have a newly created descriptor
    instance, no matter what input descriptor is provided by a user.
 
+.. c:macro:: NPY_DT_get_sort_function
+
+.. c:type:: int *(PyArrayDTypeMeta_GetSortFunction)(PyArray_Descr *, \
+        npy_intp sort_kind, PyArray_SortFuncWithContext **out_sort, \
+        NpyAuxData **out_auxdata, NPY_ARRAYMETHOD_FLAGS *out_flags)
+
+    .. versionadded:: 2.4
+
+    If defined, sets a custom sorting function for the DType for each of
+    the sort kinds numpy implements. Currently, sorts are always descending
+    and always use nulls to the end, and this must be checked in the
+    implementation. Returns 0 on success.
+
+.. c:macro:: NPY_DT_get_argsort_function
+
```

Comment:
Should macros be all caps?

Reply:
Looks like this actually a variable?

Reply:
Hmm, they seem to be, but all the dtype slots seem to be marked as macros right now.